pollination/honeybee_radiance/glare.py

```diff
@@ -0,0 +1,122 @@
+        description='Glare limit indicating presence of glare.', default=0.4
+    )
+
+    threshold_factor = Inputs.float(
+        description='Constant threshold factor in cd/m2.', default=2000
+    )
+
+    occupancy_scheulde = Inputs.file(
+        description='Path to occupancy schedule.', path='occupancy_schedule.csv',
+    )
+
+    @command
+    def run_dcglare(self):
+        return 'honeybee-radiance dcglare two-phase dc_direct.mtx dc_total.mtx ' \
+            'sky.smx view_rays.ray --glare-limit {{self.glare_limit}} ' \
+            '--threshold-factor "{{self.threshold_factor}}" --occupancy-schedule ' \
+            '{{self.occupancy_schedule}} --output occupied.ga'
+
+    dcglare_glare_autonomy = Outputs.file(
+        description='Output dcglare glare autonomy file.', path='occupied.ga'
+    )
+
+
+@dataclass
+class DCGlare(Function):
+    """Calculates DGP for all sky conditions in the sky matrix."""
+
+    dc_direct = Inputs.file(
+        description='Path to dcdirect.', path='dc_direct.mtx',
+    )
+
+    dc_total = Inputs.file(
+        description='Path to dctotal.', path='dc_total.mtx',
+    )
+
+    sky_vector = Inputs.file(
+        description='Path to sky vector.', path='sky.smx'
+    )
+
+    view_rays = Inputs.file(
+        description='Path to view ray.', path='view_rays.ray',
+    )
+
+    threshold_factor = Inputs.float(
+        description='Constant threshold factor in cd/m2.', default=2000
+    )
+
+    @command
+    def run_dcglare(self):
+        return 'honeybee-radiance dcglare two-phase dc_direct.mtx dc_total.mtx ' \
+            'sky.smx view_rays.ray --threshold-factor "{{self.threshold_factor}}" ' \
+            '--output view_rays.dgp'
+
+    view_rays_dgp = Outputs.file(
+        description='Output dcglare dgp file.', path='view_rays.dgp'
+    )
+
+
+@dataclass
+class DCGlareOccupancy(Function):
+    """Calculates DGP for all sky conditions in the sky matrix, but filtered by an
+    occupancy schedule. This means that unoccupied hours will be zero DGP."""
+
+    dc_direct = Inputs.file(
+        description='Path to dcdirect.', path='dc_direct.mtx',
+    )
+
+    dc_total = Inputs.file(
+        description='Path to dctotal.', path='dc_total.mtx',
+    )
+
+    sky_vector = Inputs.file(
+        description='Path to sky vector.', path='sky.smx'
+    )
+
+    view_rays = Inputs.file(
+        description='Path to view ray.', path='view_rays.ray',
+    )
+
+    threshold_factor = Inputs.float(
+        description='Constant threshold factor in cd/m2.', default=2000
+    )
+
+    occupancy_scheulde = Inputs.file(
+        description='Path to occupancy schedule.', path='occupancy_schedule.csv',
+    )
+
+    @command
+    def run_dcglare(self):
+        return 'honeybee-radiance dcglare two-phase dc_direct.mtx dc_total.mtx ' \
+            'sky.smx view_rays.ray --threshold-factor "{{self.threshold_factor}}" ' \
+            '--occupancy-schedule {{self.occupancy_schedule}} --output occupied.dgp'
```

Comment:
What if you make the occupancy_schedule an optional input and then in the honeybee-radiance dcglare command check and see if the schedule file is available. If it is then calculate the occupied dgp otherwise calculate the total dgp.

Here is an example of how we deal with such cases of optional file or folder inputs: https://github.com/ladybug-tools/honeybee-radiance/blob/b5436ab56f96dedd32ea093988091b3a1fec1dba/honeybee_radiance/cli/view.py#L201-L203

Reply:
Hi @mostaphaRoudsari. That makes sense. The main reason why I made three functions is because I was not sure how to handle the optional schedule. I should also get used to the output file name not being that important since we rename it in the recipes, which allows a generic name in the function.

I merged the last two functions into one.

I also changed `honeybee-radiance dcglare`:
https://github.com/ladybug-tools/honeybee-radiance/pull/714